Jakz/sign in wallet

jakzaizzat opened the pull request:
### Summary of Changes

Provide a brief description of the changes made in this PR. Explain the problem you are fixing or the feature you are adding.

### Demo or Before/After Pics

- If this is a new feature, include screenshots or recordings of the feature in action.
- If this PR makes visual changes, include before and after screenshots. You can type `/table` to quickly come up with a before/after table.

### Edge Cases

List any common edge cases that you have considered and tested.

### Testing Steps

Provide steps on how the reviewer can test the changes.

### Checklist

Please make sure to review and check all of the following:

- [ ] I've tested the changes and all tests pass.
- [ ] (if web) I've tested the changes on various desktop screen sizes to ensure responsiveness.
- [ ] (if mobile) I've tested the changes on both light and dark modes.

jakzaizzat commented on apps/mobile/src/components/Login/ConnectWalletButton.tsx:
```diff
@@ -0,0 +1,122 @@
+  icons: [
+    'https://f4shnljo4g7olt4wifnpdfz6752po37hr3waoaxakgpxqw64jojq.arweave.net/LyR2rS7hvuXPlkFa8Zc-_3T3b-eO7AcC4FGfeFvcS5M',
+  ],
+  redirect: {
+    native: 'YOUR_APP_SCHEME://',
+    universal: 'YOUR_APP_UNIVERSAL_LINK.com',
+  },
```
Where I can find the app schema url?

jakzaizzat commented on apps/mobile/src/screens/Login/LandingScreen.tsx:
```diff
@@ -16,10 +20,16 @@ import { EmailIcon } from './EmailIcon';
 import { LandingLogo } from './LandingLogo';
 import { QRCodeIcon } from './QRCodeIcon';
 
+const IS_ONBOARDING_ENABLED = true;
```
Seems like we can't used feature flag since the user didn't logged in yet, so we can't check the role. 

Replaced it with hardcode data. Lmk if there is any alternative.

Robinnnnn reviewed (commented):
you're a beasttttttttt 🔥 🔥 🔥 

some small observations

1. can we rename "Which network" to just "Network"?

<img width="225" alt="image" src="https://github.com/gallery-so/gallery/assets/12162433/ae6b5af1-ab05-496d-b787-ef309c141d71">

2. i noticed that after the wallet connect modal goes away, it hangs for a few seconds before logging you in. could we have a loading state? maybe worth asking fraser to see what that could look like!

<img width="322" alt="image" src="https://github.com/gallery-so/gallery/assets/12162433/812e7196-009d-48f4-988c-4b813c288ebd">

jakzaizzat commented:
@Robinnnnn Thanks ser!

1. Renamed it
2. Added loading state to the login screen while waiting for new design for (signature stuff)


https://github.com/gallery-so/gallery/assets/4480258/62254602-a45d-44b2-9bcf-c65c5f2344fc

jakzaizzat commented:
Update the loading state


https://github.com/gallery-so/gallery/assets/4480258/49ac287e-9371-4986-ba72-e1b2247e15ce

kaitoo1 commented on apps/mobile/src/hooks/useLogout.ts:
```diff
@@ -27,6 +29,7 @@ export function useLogout(): [() => void, boolean] {
   const logout = useCallback(async () => {
     try {
       magic.user.logout();
+      provider?.disconnect();
```
nice catch

kaitoo1 commented on apps/mobile/src/icons/EmailIcon.tsx:
```diff
@@ -0,0 +1,10 @@
+import Svg, { Path, SvgProps } from 'react-native-svg';
```
oo we have an EmailIcon already, looks like it's in screens/Login/EmailIcon

I think this /icons/ folder is better though, would you be able to consolidate these?

jakzaizzat replied:
Yup! I've deleted the new files and consolidate the icon into the `icons` folder

kaitoo1 commented on apps/mobile/src/components/Login/WalletSelectorBottomSheet.tsx:
```diff
@@ -0,0 +1,142 @@
+
+export { ForwardedWalletSelectorBottomSheet as WalletSelectorBottomSheet };
+
+function EvmStackedIcons() {
```
small design comment, on dark mode we want the border around each icon to match the background - basically appear transparent

<img width="95" alt="CleanShot 2023-08-28 at 21 33 59@2x" src="https://github.com/gallery-so/gallery/assets/80802871/fd6b5135-4278-4235-90f1-189b9b370284">


design:
<img width="827" alt="CleanShot 2023-08-28 at 21 41 40@2x" src="https://github.com/gallery-so/gallery/assets/80802871/9a9edf92-6249-4db0-98b3-587fb95f1786">

jakzaizzat replied:
Updated on dark mode version!

<img width="417" alt="CleanShot 2023-08-29 at 08 46 28@2x" src="https://github.com/gallery-so/gallery/assets/4480258/dd9a0f5b-f401-458b-9184-27be12c4c134">

kaitoo1 commented on apps/mobile/src/components/Login/WalletSelectorBottomSheet.tsx:
```diff
@@ -0,0 +1,142 @@
+                paddingBottom: bottom,
+              }}
+            >
+              <SpinnerIcon spin />
```
looks like we  need a dark mode version 😅
https://github.com/gallery-so/gallery/assets/80802871/b5adf765-2c47-4d53-b253-e8ac03aa845c

jakzaizzat replied:
lololo. Updated on dark mode version 🤦‍♂️


https://github.com/gallery-so/gallery/assets/4480258/77606d7b-3840-4255-a3e7-8d5122e09e7f

kaitoo1 commented:
jakz lets gooo!!! this is sick, looking great.

left a couple of smaller comments about design and file placement, code looks great.

also noticed a few improvements we can probably make for when there's an error or user cancels the flow:
I closed the bottom sheet while it was waiting for me to sign a message, and now my app is stuck in the loading state. even if I restart the app, it automatically goes into the loading state, so I can't re-attempt the flow.

attaching a video of it automatically showing the loading spinner when I open the app.
https://github.com/gallery-so/gallery/assets/80802871/3a76406f-efe5-4047-8a2a-33e6b2a99713

I think some ways we can handle this are
- if the user re-launches the app, reset the wallet connect state
- if the user closes the bottom sheet during the sign in flow, reset the wallet connect state

jakzaizzat commented:
@kaitoo1  Added the `onDismiss` callback to reset the wallet connection state


https://github.com/gallery-so/gallery/assets/4480258/97a83c63-5040-4c94-a3bf-54754c482758

kaitoo1 commented:
nice jakz!!!! thanks for addressing comments, it looks great!
